Dockerfile

```diff
@@ -1,27 +1,20 @@
-FROM ubuntu:22.04 as builder
+FROM golang:1.19-bullseye as builder
 
 RUN apt-get update && \
-    apt-get install -y --no-install-recommends curl git ca-certificates build-essential libelf-dev gnupg2
+    apt-get install -y --no-install-recommends build-essential libelf-dev gnupg2
 
-RUN echo 'deb https://ppa.launchpadcontent.net/longsleep/golang-backports/ubuntu jammy main' > /etc/apt/sources.list.d/golang-backports.list && \
-    apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 52b59b1571a79dbc054901c0f6bc817356a3d45e && \
-    apt-get update && \
-    apt-get install -y --no-install-recommends golang-1.19-go
-
-RUN mkdir /build
```

Comment:
The number of layers doesn't really matter here as we only use the first image for building. None of the layers make it into the final image.

That said, the build steps are pretty fast and your grouping seems logical.